consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validator/ForkingValidatorProvider.java

```diff
@@ -95,8 +95,13 @@ public ForkingValidatorProvider(
         final long prevBlockNumber = blockNumber - 1L;
         final Optional<BlockHeader> prevBlockHeader = blockchain.getBlockHeader(prevBlockNumber);
         if (prevBlockHeader.isPresent()) {
-          return resolveValidatorProvider(prevBlockNumber)
-              .getValidatorsForBlock(prevBlockHeader.get());
+          final Collection<Address> validatorsForPreviousBlock =
+              resolveValidatorProvider(prevBlockNumber)
+                  .getValidatorsForBlock(prevBlockHeader.get());
+          // Update VoteTallyCache
+          blockValidatorProvider.setValidatorsForBlock(
+              prevBlockHeader.get(), validatorsForPreviousBlock);
```

Comment:
Really, we want to be setting the validatorsForPreviousBlock against the yet to be created block, so it might make sense to update the cache once the block is created/imported